File: warrior/src/main/java/com/dumbdogdiner/warrior/commands/warrior/WarriorLobbyCommand.java
package com.dumbdogdiner.warrior.commands.warrior;

import com.dumbdogdiner.warrior.api.command.SubCommand;
import com.dumbdogdiner.warrior.managers.LobbyManager;
import com.dumbdogdiner.warrior.api.util.TranslationUtil;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import java.util.List;

public class WarriorLobbyCommand implements SubCommand {
    @Override
    public String getAlias() {
        return "setlobby";
    }

    @Override
    public String getSyntax() {
        return "/warrior setlobby";
    }

    @Override
    public String getPermission() {
        return "warrior.command.setlobby";
    }

    @Override
    public boolean execute(CommandSender sender, String commandLabel, String[] args) {
        LobbyManager.updateLocation(((Player)sender).getLocation());
        sender.sendMessage("Set Lobby Spawn to " + TranslationUtil.readableLocation(((Player) sender).getLocation(), true, true));
        return true;
    }

    @Override
    public List<String> getArguments(CommandSender sender, String[] arguments) {
        return null;
    }
}

File: warrior/src/main/kotlin/com/dumbdogdiner/warrior/commands/warrior/WarriorLobbyCommand.kt
package com.dumbdogdiner.warrior.commands.warrior

import com.dumbdogdiner.warrior.Warrior
import com.dumbdogdiner.warrior.api.command.SubCommand
import com.dumbdogdiner.warrior.api.util.TranslationUtil
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

class WarriorLobbyCommand : SubCommand {
    override fun getAlias(): String {
        return "setlobby"
    }

    override fun getSyntax(): String {
        return "/warrior setlobby"
    }

    override fun getPermission(): String {
        return "warrior.command.setlobby"
    }

    override fun execute(sender: CommandSender, commandLabel: String, args: Array<String>): Boolean {
        Warrior.lobbyManager.updateLocation((sender as Player).location)
        sender.sendMessage(
            "Set Lobby Spawn to " + TranslationUtil.readableLocation(
                sender.location,
                true,
                true
            )
        )
        return true
    }

    override fun getArguments(sender: CommandSender, arguments: Array<String>): List<String> {
        return listOf()
    }
}